### PR TITLE
coprocessor/dag: fix bug caused by check meta in chunk

### PR DESCRIPTION
**Pull request by AndreMouche**
@BusyJay @hicqu PTAL

### CODE REVIEW

File: src/coprocessor/dag/dag.rs [hunk 1]
```diff
@@ -55,18 +57,26 @@ impl<'s> DAGContext<'s> {
...
             req_ctx: req_ctx,
+            batch_row_count: batch_row_count,
         }
     }
 
     pub fn handle_request(mut self, statistics: &'s mut Statistics) -> Result<Response> {
         self.validate_dag()?;
         let mut exec = self.build_dag(statistics)?;
         let mut chunks = vec![];
+        let mut cur_chunk_cnt = 0;
```
Comment (BusyJay) [inline src/coprocessor/dag/dag.rs]: I think it should be `record_cnt`.

**Comment by BusyJay** [comment]
LGTM, /run-integration-tests

File: src/coprocessor/endpoint.rs [hunk 2]
```diff
@@ -645,19 +647,42 @@ impl TiDbEndPoint {
...
+        batch_row_limit: usize,
+    ) -> Result<Response> {
+        let ctx = SelectContext::new(
+            sel,
+            self.snap.as_ref(),
+            &mut t.statistics,
+            &t.ctx,
+            batch_row_limit,
+        )?;
         let range = t.req.get_ranges().to_vec();
```
Comment (hicqu) [inline src/coprocessor/endpoint.rs]: `take_ranges.into_vec()` is better.

**Comment by hicqu** [comment]
Rest LGTM.

**Pull request merged**